pkg/auth/auth.go

```diff
@@ -69,30 +70,52 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 	systemContext = systemContextWithOptions(systemContext, opts.AuthFile, opts.CertDir)
 
 	var (
-		server string
-		err    error
+		key string
+		err error
 	)
-	if len(args) > 1 {
-		return errors.New("login accepts only one registry to login to")
-	}
-	if len(args) == 0 {
+	l := len(args)
+	switch l {
+	case 0:
 		if !opts.AcceptUnspecifiedRegistry {
 			return errors.New("please provide a registry to login to")
 		}
-		if server, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
+		if key, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
 			return err
 		}
-		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", server)
-	} else {
-		server = getRegistryName(args[0])
+		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", key)
+
+	case 1:
+		key = args[0]
+
+	default:
+		return errors.New("login accepts only one registry to login to")
+
 	}
-	authConfig, err := config.GetCredentials(systemContext, server)
-	if err != nil {
-		return errors.Wrap(err, "reading auth file")
+
+	var (
+		authConfig types.DockerAuthConfig
+		domain     string
+	)
+	if opts.AcceptRepositories {
+		authConfig, domain, err = getCredentialsForRef(systemContext, key)
```

Comment:
RFC: Should `podman login --accept-repositories quay.io` (i.e. registry-only) be allowed?

I’m leaning towards “yes” (and eventually, or in new tools, we could make `--accept-repositories` the default, and reject URLs), but I can see the case for only using the option for namespaced keys.

Reply:
> RFC: Should `podman login --accept-repositories quay.io` (i.e. registry-only) be allowed?

Yeah I think so.